app/bundles/LeadBundle/Entity/LeadListRepository.php

```diff
@@ -1677,11 +1677,44 @@ public function getListFilterExpr($filters, &$parameters, QueryBuilder $q, $isNo
 
                         case 'in':
                         case 'notIn':
+                            if (is_string($details['filter'])) {
+                                // Expound the string to an array based on common string array notations.
+                                foreach ([',', ';', "\t", ' '] as $delimiter) {
```

Comment:
I don't like the idea of spaces as delimiters. Users will fill in value like `New York, Boston`. Building arrays from a text value is used elsewhere in Mautic too. For for example defining multiple To/CC/BCC email addresses. Only commas are used as separators in all those places. Do you think you could remove the other delimiters?

Then you could use simpler parsing by using [str_getcsv](http://php.net/manual/en/function.str-getcsv.php) function.

Reply:
These are "possible" delimiters. So it is compatible with str_getcsv, spaces only being the last possible option. I was trying to support all the different kinds of spreadsheet apps here... 
That being said this should be done by JS upon pasting in values, not in the PHP backend. That way there would be no question how the separate values would be treated.

I think I'll refactor this PR today.